sql/V001__Initial_Setup.sql

```diff
@@ -647,6 +658,23 @@ CREATE TABLE step_science (
 
 ALTER TABLE step_science OWNER TO postgres;
 
+
+--
+-- Data for Name: dataset; Type: TABLE DATA; Schema: public; Owner: postgres
+--
+
+CREATE TABLE dataset (
+    dataset_label   character varying(46)        PRIMARY KEY,
+    observation_id  character varying(40)        REFERENCES observation ON DELETE CASCADE,
+    dataset_index   id_index                     NOT NULL,
+    step_id         integer                      REFERENCES step        ON DELETE CASCADE,
+    filename        character varying(24)        NOT NULL,
+    "timestamp"     timestamp (5) WITH TIME ZONE NOT NULL
+    CONSTRAINT dataset_id_check CHECK ((dataset_label)::text = (((observation_id)::text || '-'::text) || trim(leading from to_char(dataset_index, '9000'))))
```

Comment:
`trim(leading from to_char(dataset_index, '9000'))` is a bit weird, admittedly, but `to_char(dataset_index, '000')` fails if there are a 1000 or more datasets and I believe we have a few such observations.